### DISCUSSION TITLE
PAT Advanced 1002. A+B for Polynomials (25) (C语言实现) | PAT Solution in C

### DISCUSSION THREAD

**Discussion by xlucn** [xlucn/PAT#59] [General]
https://oliverlew.github.io/PAT/Advanced/1002.html 

 题目

**Comment by celiae** [comment]
while里已经判断过i<KA,为啥if里又写一遍。。。为什么不写fabs就不能通过所有测试

**Comment by xlucn** [comment]
> @celiae

时间久远，可能不很清楚了，尝试解释一下：

> while里已经判断过i<KA,为啥if里又写一遍。。。

While的判断是两个条件“或”的结果，可能其中一个条件不满足。第一个if不就有`i == KA`的情况嘛。

> 为什么不写fabs就不能通过所有测试

要考虑负数呀，比如 -x ，系数 -1 小于 0.05，但是要输出来。
